### PR TITLE
Movie list api

### PR DESCRIPTION
**Pull request by harshup20**
1.removed the unused functions, classes, fields 
2. solved the issues occurring while running the application

### CODE REVIEW

File: lib/datasource/entity/maps_extensions.dart [hunk 1]
```diff
@@ -13,4 +13,10 @@ extension DefaultMap<K, V> on Map<K, V> {
       return defaultValue;
     }
   }
+}
+
+
+
+class xyz{
```
Comment (mayyuu) [inline lib/datasource/entity/maps_extensions.dart:20]: Can you please remove this unwanted class

File: lib/datasource/remote/imlementation/movies_datasource_impl.dart [hunk 2]
```diff
@@ -41,13 +44,44 @@ class MoviesDataSourceImpl extends MovieListDataSource {
       throw MovieLandingError(
           type: MovieLandingErrorType.SERVER_MESSAGE);
     }
-    throw UnimplementedError();
+    throw MovieLandingError(
+        type: MovieLandingErrorType.SERVER_MESSAGE);
   }
+*/
+  // @override
```
Comment (mayyuu) [inline lib/datasource/remote/imlementation/movies_datasource_impl.dart:51]: Can you please remove unwanted comments everywhere from the code

File: lib/datasource/remote/imlementation/movies_datasource_impl.dart [hunk 3]
```diff
@@ -41,13 +44,44 @@ class MoviesDataSourceImpl extends MovieListDataSource {
...
   @override
-  Future<List<MovieTrailerDomain>> getMovieTrailer(GetMovieTrailerUseCaseParams params) {
-    // TODO: implement getMovieTrailer
-    throw UnimplementedError();
+  Future<List<MovieDomain>> getPopularMovie(
+      {GetPopularMovieUseCaseParams params}) async{
+    var response = await _movieRequest.getPopularMovie(params: params);
+    MovieResponse movieResponse = MovieResponse(response);
+    if (movieResponse.getErrors().length != 0) {
+      BaseError baseError = movieResponse.getErrors().first;
```
Comment (mayyuu) [inline lib/datasource/remote/imlementation/movies_datasource_impl.dart:65]: Can you please check and handel the errors into data source impl class

File: lib/datasource/remote/providers/rest/dto/movie_landing_error_dto.dart [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
   MovieLandingErrorDto({this.errors});
 
   MovieLandingErrorDto.fromJson(Map<String, dynamic> json) {
     List<dynamic> errorsList = json['errors'];
     this.errors = [];
-    this.errors!.addAll(errorsList.map((o) => o.toString()));
+    this.errors.addAll(errorsList.map((o) => o.toString()));
   }
 
   Map<String, dynamic> toJson() {
```
Comment (mayyuu) [inline lib/datasource/remote/providers/rest/dto/movie_landing_error_dto.dart:12]: Can we please the need and implementation  of this class,and if its not used can we please remove it

File: lib/datasource/remote/providers/rest/request/movie_landing_request.dart [hunk 5]
```diff
@@ -24,11 +22,11 @@ abstract class MovieLandingRestRequest extends RESTRequest {
   }
 
   @override
-  Future<Response> execute(String endpoint, Map<String, dynamic> params,
+  Future<Response> execute(dynamic page, Map<String, dynamic> params,
       int apiCallMethod, int apiIdentifier,
       {forceRefresh: false}) async {
     return await super
-        .execute(endpoint, params, apiCallMethod, apiIdentifier,
+        .execute(page.toString(), params, apiCallMethod, apiIdentifier,
```
Comment (mayyuu) [inline lib/datasource/remote/providers/rest/request/movie_landing_request.dart:29]: Can you please explain why we have used page here

File: lib/movie_landing_page_router.dart [hunk 6]
```diff
@@ -1,26 +1,31 @@
...
           type: PageTransitionType.fade,
           duration: Duration(milliseconds: 450),
         );
       break;
       case MovieLandingRoutePaths.Movie:
         return PageTransition(
-          // child: ContactView(),
-          child: AppLauncher(url: '', trailerName: '',),
+          child: MovieDetailScreen(movieItem: settings.arguments,),
+          // child: AppLauncher(url: '', trailerName: '',),
```
Comment (mayyuu) [inline lib/movie_landing_page_router.dart:28]: Please remove commented code from all files.

File: lib/presentation/movie_list/widgets/movie_detail_description.dart [hunk 7]
```diff
@@ -1,24 +1,26 @@
 import 'package:flutter/material.dart';
 
 import 'package:newfluttermovieapp/domain/model/movie_domain.dart';
+import 'package:newfluttermovieapp/presentation/model/movie_item.dart';
 
 import 'app_state_container.dart';
 
 
 class MovieDetailDescription extends StatelessWidget {
```
Comment (mayyuu) [inline lib/presentation/movie_list/widgets/movie_detail_description.dart:9]: Not sure why have we taken basestatefulll widget everytime can you please explain

File: lib/datasource/remote/providers/rest/dto/movie_landing_error_dto.dart (hunk 4 above)
Comment (harshup20) [reply lib/datasource/remote/providers/rest/dto/movie_landing_error_dto.dart:12]: its used in the response class 
 if (super.response.statusCode != 200) {
      MovieLandingErrorDto errorDto =
      MovieLandingErrorDto.fromJson(response as Map<String, dynamic>);
      getErrors().add(BaseError(
          message: errorDto.errors?.first.toString(),
          type: MovieLandingErrorType.SERVER_MESSAGE));

File: lib/datasource/datasources/movies_datasource.dart [hunk 8]
```diff
@@ -5,6 +5,7 @@ import 'package:newfluttermovieapp/domain/usecase/get_movie_trailer_usecase.dart
 import 'package:newfluttermovieapp/domain/usecase/get_popular_movie_usecase.dart';
 
 abstract class MovieListDataSource{
```
Comment (mayyuu) [inline lib/datasource/datasources/movies_datasource.dart:7]: Can You please update MovieListDatasource to MovieDatasource

File: lib/datasource/remote/imlementation/movies_datasource_impl.dart [hunk 9]
```diff
@@ -1,64 +1,24 @@
...
 import 'package:newfluttermovieapp/datasource/remote/providers/rest/response/movie_response.dart';
+import 'package:newfluttermovieapp/datasource/remote/providers/rest/response/trailer_response.dart';
 import 'package:newfluttermovieapp/domain/model/movie_domain.dart';
 import 'package:newfluttermovieapp/domain/model/movie_trailer_domain.dart';
 import 'package:newfluttermovieapp/domain/usecase/get_movie_trailer_usecase.dart';
 import 'package:newfluttermovieapp/domain/usecase/get_popular_movie_usecase.dart';
 import 'package:newfluttermovieapp/presentation/base/view/movie_landing_base_view.dart';
+import 'package:newfluttermovieapp/datasource/entity/trailer_entity.dart';
 
 class MoviesDataSourceImpl extends MovieListDataSource {
```
Comment (naikdp7) [inline lib/datasource/remote/imlementation/movies_datasource_impl.dart:14]: which datasource include it in the className

File: lib/datasource/entity/trailer_entity.dart [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+      trailerKey: json.getOrElse('key', ''),
+      trailerName: json.getOrElse('name', ''),
+      trailerSite: json.getOrElse('site', ''),
+      trailerSize: json.getOrElse('size', 0),
+
+    );
+  }
+}
+
+extension MovieEntityExtention on TrailerEntity {
```
Comment (mayyuu) [inline lib/datasource/entity/trailer_entity.dart:28]: can you update TrailerEntityExtension name as the extension is related to trailer class

File: lib/datasource/entity/trailer_entity.dart [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  TrailerDomain mapTrailerToDomain() => TrailerDomain(
+   trailerKey:trailerKey,
+    trailerName: trailerName,
+    trailerSite: trailerSite,
+    trailerSize: trailerSize
+  );
+}
+extension MovieListExtension on List<TrailerEntity> {
```
Comment (mayyuu) [inline lib/datasource/entity/trailer_entity.dart:36]: Same here

File: lib/datasource/remote/providers/rest/request/movie_request.dart [hunk 12]
```diff
@@ -1,31 +1,52 @@
...
+    Map<String, dynamic> params = Map();
+    params.putIfAbsent("api_key", () {
+      return '802b2c4b88ea1183e50e6b285a27696e';
+    });
+    params.putIfAbsent("language", () {
+      return 'en-US';
+    });
+    params.putIfAbsent("page", () {
+      return '1';
+    });
```
Comment (mayyuu) [inline lib/datasource/remote/providers/rest/request/movie_request.dart:28]: Can you please tell the use of GetPopularMovieUseCaseParams here , as i can see we have provided but have not used it.

File: lib/datasource/remote/providers/rest/request/movie_request.dart [hunk 13]
```diff
@@ -1,31 +1,52 @@
...
+      return '802b2c4b88ea1183e50e6b285a27696e';
+    });
+    paramsMap.putIfAbsent("language", () {
+      return 'en-US';
+    });
+    paramsMap.putIfAbsent("page", () {
+      return '1';
+    });
+    print('the end point trailer${Endpoint.MOVIE_TRAILER}');
+    return await execute(params.movieId.toString()+Endpoint.MOVIE_TRAILER, paramsMap,
```
Comment (mayyuu) [inline lib/datasource/remote/providers/rest/request/movie_request.dart:47]: Same here

File: lib/movie_landing_page_application_viewmodel.dart [hunk 14]
```diff
@@ -1,91 +1,91 @@
...
+//     } else {
+//       lokoTheme = LokoTheme.dark;
+//     }
+//   }
+// }
+//
+// enum LokoTheme {
+//   dark,
+//   light,
+// }
```
Comment (mayyuu) [inline lib/movie_landing_page_application_viewmodel.dart:91]: Please remove this if its not needing

File: lib/movie_landing_page_router.dart [hunk 15]
```diff
@@ -1,27 +1,31 @@
...
-  static Route<dynamic>? generateRoute(RouteSettings settings) {
+
+  static Route<dynamic> generateRoute(RouteSettings settings) {
     switch (settings.name) {
       case MovieLandingRoutePaths.Landing:
+        // MovieListParams params;
         return PageTransition(
-          child: Container(),
-          // child: LandingView(),
+          // child: Container(),
```
Comment (mayyuu) [inline lib/movie_landing_page_router.dart:18]: Please remove commented code

File: lib/presentation/movie_list/movie_detail_screen.dart [hunk 16]
```diff
@@ -0,0 +1,204 @@
...
+    return _scrollController.hasClients &&
+        _scrollController.offset > (200 - kToolbarHeight);
+  }
+
+
+  // Link:- https://flutter-widget.live/widgets/NestedScrollView
+
+  @override
+  Widget buildBody() {
+    Scaffold(
```
Comment (mayyuu) [inline lib/presentation/movie_list/movie_detail_screen.dart:43]: Please refer comtact_view_web file for the same as you can override build method there

File: lib/datasource/remote/providers/rest/request/movie_landing_request.dart (hunk 5 above)
Comment (harshup20) [reply lib/datasource/remote/providers/rest/request/movie_landing_request.dart]: accepting the page as a parameter for changing the page

File: lib/datasource/remote/imlementation/movies_datasource_impl.dart (hunk 3 above)
Comment (harshup20) [reply lib/datasource/remote/imlementation/movies_datasource_impl.dart:25]: I am unclear with the error handling class please explain

**Review by mayyuu — APPROVED** [review]

**Pull request merged**